faunadb/client_test.go

```diff
@@ -2382,6 +2382,112 @@ func (s *ClientTestSuite) TestEchoQuery() {
 	s.Require().True(firstSeen > 0 && s.client.GetLastTxnTime() >= firstSeen)
 }
 
+func (s *ClientTestSuite) TestPaginateAccessProviders() {
+	var obj f.ObjectV
+	var val f.Value
+	var arr f.Arr
+	var b bool
+
+	name := f.RandomStartingWith("name_")
+	issuer := f.RandomStartingWith("https://xxxx.auth0.com")
+	jwksUri := f.RandomStartingWith("https://xxxx.auth0.com/")
+	//allowed_roles := f.Arr{}
+	//allowed_collections := f.Arr{}
+
+	query := f.CreateAccessProvider(f.Obj{
+		"name":     name,
+		"issuer":   issuer,
+		"jwks_uri": jwksUri,
+	})
+	s.adminQueryAndDecode(query, &obj)
+	obj.At(refField).Get(&ref)
+
+	s.adminQueryAndDecode(f.Paginate(f.AccessProviders()), &val)
+	val.At(dataField).Get(&arr)
+
+	s.Require().True(len(arr) != 0)
+
+	s.adminQueryAndDecode(f.GT(f.Count(f.AccessProviders()), 0), &b)
+	s.Require().True(b)
```

Comment:
Asserting quantities is a weak type of assertion. I'd recommend asserting the value returned is equal to what you've just created.